src/components/date/date.jsx

```diff
@@ -0,0 +1,264 @@
+import PropTypes from 'prop-types';
+import React, {
+  forwardRef,
+  useEffect,
+  useLayoutEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from 'react';
+import calendarSvg from '../../svgs/calendar.svg';
+import cautionSvg from '../../svgs/caution.svg';
+import Calendar from '../calendar/calendar.jsx';
+import FormControl from '../form-control/form-control.jsx';
+import IconButton from '../icon-button/icon-button.jsx';
+import Icon from '../icon/icon.jsx';
+import useDropdownClose from '../../hooks/use-dropdown-close.js';
+import styles from './date.css';
+
+/* eslint-disable react/prop-types */
+function ControlIcons(props) {
+  return (
+    <div className={styles.icons}>
+      {!!props.validationMessage && (
+        <Icon
+          className={styles['invalid-icon']}
+          icon={cautionSvg}
+          label={props.validationMessage}
+        />
+      )}
+      {props.readOnly ? (
+        <Icon
+          icon={calendarSvg}
+          title={props.label}
+          label={`${props.label} calendar icon`}
+        />
+      ) : (
+        <IconButton
+          onPress={props.onPress}
+          icon={calendarSvg}
+          title={props.label}
+          label={`${props.label} calendar button`}
+        />
+      )}
+    </div>
+  );
+}
+/* eslint-enable react/prop-types */
+
+function toDateStringFormat(date) {
+  return date ? date.toLocaleDateString(undefined, { month: 'short', year: 'numeric', day: 'numeric' }) : '';
+}
+
+function toFullDateFormat(date) {
+  return date ? date.toISOString().slice(0, 10) : undefined;
+}
+
+function fromFullDateFormat(string) {
+  const date = new window.Date(`${string}T00:00:00`);
+  if (date.toDateString() === 'Invalid Date') {
+    return undefined;
+  }
+
+  return date;
+}
+
+const Date = forwardRef(function Date(props, forwardedRef) {
+  const backupRef = useRef();
+  const ref = forwardedRef || backupRef;
+  const containerRef = useRef();
+  const inputRef = useRef();
+  const [active, setActive] = useState(false);
+  const [editing, setEditing] = useState(!!props.validationMessage);
+  const [expanded, setExpanded] = useState(false);
+  const [validationMessage, setValidationMessage] = useState(props.validationMessage);
+  const [value, setValue] = useState(fromFullDateFormat(props.value));
+  const classNames = {
```

Comment:
I'm trying out a new CSS API for our components. Normally, I would prefer individual props prefixed with css (e.g. `cssContainer`, `cssCalendarContainer`, and `cssInput`). 

However, I do see that most of our components also have `className`. I think maybe the compromise is `className` _or_ `classNames`. Most of our complex components will need multiple classNames anyways. Though, that does imply features will be introducing one-offs (something to discourage).

Reply:
For me, `className` has always been about keeping parity with the basic React API, where this sets the root element class.

Reply:
I do very much like this new pattern, though. Makes it easy to see what all of the classes actually are without needing to go into the css file.

Reply:
> For me, `className` has always been about keeping parity with the basic React API, where this sets the root element class.

Yeah, same. I think the `className` or `classNames` might be a good pattern to avoid too much mental overhead. Who knows maybe we just use `className` as a string or an object? 🤔